pkg/loadbalancer/server.go

```diff
@@ -163,7 +163,7 @@ func (s *Server) EnsureLoadBalancerDeleted(ctx context.Context, clusterName stri
 func loadBalancerName(clusterName string, service *v1.Service) string {
 	hash := sha256.Sum256([]byte(loadBalancerSimpleName(clusterName, service)))
 	encoded := base32.StdEncoding.EncodeToString(hash[:])
-	name := constants.ContainerPrefix + "-" + encoded[:40]
```

Comment:
let's print the first 6 hash bytes that is more than enough to avoid collisions on names 
https://en.wikipedia.org/wiki/Birthday_problem

See https://en.wikipedia.org/wiki/Birthday_problem

```go
        h := sha256.New()
	io.WriteString(h, loadBalancerSimpleName(clusterName, service))
	hash := h.Sum(nil)
	return fmt.Sprintf("%s-%x", constants.ContainerPrefix, hash[:6])
```

Reply:
you also need to update the test in with the new name

https://github.com/kubernetes-sigs/cloud-provider-kind/blob/bdc116dd8bf7cb3c9dbc69fd8fb83974c6c2d321/pkg/loadbalancer/server_test.go#L11-L38